src/java/voldemort/store/readonly/ReadOnlyStorageEngine.java

```diff
@@ -414,7 +414,7 @@ public void run() {
                     Utils.rm(file);
                     logger.info("Deleting of " + file.getAbsolutePath()
                                 + " completed successfully.");
-                    storeVersionManager.syncInternalStateFromFileSystem();
+                    storeVersionManager.syncInternalStateFromFileSystem(true);
```

Comment:
In my opinion, it would have been easier to add an another method call here, to remove the HDFS state, instead of the boolean flag.

This is a nitpick, so no need to change the code.